src/Model/Payload.php

```diff
@@ -113,4 +110,39 @@ public function getCustomData(): array
     {
         return $this->customData;
     }
+
+    /**
+     * Guard the custom data upon instantiation.
+     *
+     * @param array $data
+     */
+    private function guardCustomData(array $data)
```

Comment:
Please rename to `validateCustomData`, because the `guard` best related to 'security'.